packages/create-project/src/create-project.ts

```diff
@@ -0,0 +1,188 @@
+import { makeDir } from './helpers/make-dir';
+import { tryGitInit } from './helpers/git';
+import { install } from './helpers/install';
+import { isFolderEmpty } from './helpers/is-folder-empty';
+import { getOnline } from './helpers/is-online';
+// import { shouldUseYarn } from './helpers/should-use-yarn';
+import { isWriteable } from './helpers/is-writeable';
+
+export class DownloadError extends Error {}
+
+export async function createApp({
+  appPath,
+  useYarn,
+  example,
+  examplePath,
+}: {
+  appPath: string;
+  useYarn: boolean;
+  example: string;
+  examplePath?: string;
+}): Promise<void> {
+  let repoInfo: RepoInfo | undefined;
+
+  if (example) {
+    let repoUrl: URL | undefined;
+
+    try {
+      repoUrl = new URL(example);
+    } catch (error: any) {
+      if (error.code !== 'ERR_INVALID_URL') {
+        console.error(error);
+        process.exit(1);
+      }
+    }
+
+    if (repoUrl) {
+      if (repoUrl.origin !== 'https://github.com') {
+        console.error(
+          `Invalid URL: ${chalk.red(
+            `"${example}"`
+          )}. Only GitHub repositories are supported. Please use a GitHub URL and try again.`
+        );
+        process.exit(1);
+      }
+
+      repoInfo = await getRepoInfo(repoUrl, examplePath);
+
+      if (!repoInfo) {
+        console.error(`Found invalid GitHub URL: ${chalk.red(`"${example}"`)}. Please fix the URL and try again.`);
+        process.exit(1);
+      }
+
+      const found = await hasRepo(repoInfo);
+
+      if (!found) {
+        console.error(
+          `Could not locate the repository for ${chalk.red(
+            `"${example}"`
+          )}. Please check that the repository exists and try again.`
+        );
+        process.exit(1);
+      }
+    } else if (example !== '__internal-testing-retry') {
+      const found = await hasExample(example);
+
+      if (!found) {
+        console.error(
+          `Could not locate an example named ${chalk.red(`"${example}"`)}. It could be due to the following:\n`,
+          `1. Your spelling of example ${chalk.red(`"${example}"`)} might be incorrect.\n`,
+          `2. You might not be connected to the internet.`
+        );
+        process.exit(1);
+      }
+    }
+  }
+
+  const root = path.resolve(appPath);
+
+  if (!(await isWriteable(path.dirname(root)))) {
+    console.error('The application path is not writable, please check folder permissions and try again.');
+    console.error('It is likely you do not have write permissions for this folder.');
+    process.exit(1);
+  }
+
+  const appName = path.basename(root);
+
+  await makeDir(root);
+  if (!isFolderEmpty(root, appName)) {
+    process.exit(1);
+  }
+
+  const isOnline = !useYarn || (await getOnline());
+  const originalDirectory = process.cwd();
+
+  const displayedCommand = useYarn ? 'yarn' : 'npm';
+  console.log(`Creating a new Temporal project in ${chalk.green(root)}/.`);
+  console.log();
+
+  await makeDir(root);
+  process.chdir(root);
```

Comment:
I usually prefer not to rely on being in a specific directory, it doesn't work when you want to run things concurrently.